stag-library-compiler/src/main/java/com/vimeo/stag/processor/generators/TypeAdapterGenerator.java

```diff
@@ -324,6 +332,76 @@ private static void runIfAnnotationSupported(@NotNull List<? extends AnnotationM
+            for (int i = 0; i < constructorParameters.size(); i++) {
+                constructorParameterStr += constructorParameters.get(i);
+                if (i != constructorParameters.size() - 1) {
+                    constructorParameterStr += ",";
+                }
+            }
+            constructorParameterStr += ")";
+            fieldAdapterAccessor += constructorParameterStr;
+        } else if (jsonAdapterType == TypeUtils.JsonAdapterType.TYPE_ADAPTER_FACTORY) {
+            TypeName typeTokenField = ParameterizedTypeName.get(ClassName.get(TypeToken.class), TypeVariableName.get(fieldType));
+            fieldAdapterAccessor += "().create(gson, new " + typeTokenField + "(){})";
+        } else if (jsonAdapterType == TypeUtils.JsonAdapterType.JSON_SERIALIZER
+                || jsonAdapterType == TypeUtils.JsonAdapterType.JSON_DESERIALIZER
+                || jsonAdapterType == TypeUtils.JsonAdapterType.JSON_SERIALIZER_DESERIALIZER) {
+            String serializer = null, deserializer = null;
+
+            if(jsonAdapterType == TypeUtils.JsonAdapterType.JSON_SERIALIZER_DESERIALIZER){
+                String varName = keyFieldName + "SerializerDeserializer";
+                String initializer = adapterType.getEnclosingElement().toString() + " " + varName + " = " +
+                        "new " + adapterType;
+                constructorBuilder.addStatement(initializer);
+                serializer = deserializer = varName;
+            }else if(jsonAdapterType == TypeUtils.JsonAdapterType.JSON_SERIALIZER){
+                serializer = "new "+ adapterType;
+            }else if(jsonAdapterType == TypeUtils.JsonAdapterType.JSON_DESERIALIZER){
+                deserializer = "new "+ adapterType;
+            }
+            TypeName typeTokenField = ParameterizedTypeName.get(ClassName.get(TypeToken.class), TypeVariableName.get(fieldType));
+            fieldAdapterAccessor = "new "+TypeVariableName.get(TreeTypeAdapter.class) + "(" + serializer +", "+ deserializer + ", gson, new "+typeTokenField + "(){}, null)";
+        } else {
+            throw new IllegalArgumentException(
+                    "@JsonAdapter value must be TypeAdapter, TypeAdapterFactory, "
+                            + "JsonSerializer or JsonDeserializer reference.");
+        }
+        //Add this to a member variable
+        String fieldName = TYPE_ADAPTER_FIELD_PREFIX + adapterFieldInfo.size();
+        String originalFieldName = FileGenUtils.unescapeEscapedString(fieldName);
+        TypeName typeName = getAdapterFieldTypeName(fieldType);
+        adapterBuilder.addField(typeName, originalFieldName, Modifier.PRIVATE, Modifier.FINAL);
+        String statement = fieldName + " = " + getCleanedFieldInitializer(fieldAdapterAccessor);
+        if (isNullSafe) {
+            statement += ".nullSafe()";
+        }
+        constructorBuilder.addStatement(statement);
+        fieldAdapterAccessor = fieldName;
+
+        return fieldAdapterAccessor;
```

Comment:
Reassigning the `fieldAdapterAccessor` variable to the `fieldName` variable makes this code a little confusing, we should just return the `fieldName` variable directly instead of reassigning. Also, if this method is called `getAdapterAccessorForKnownJsonAdapterType` why is it returning a field name? It seems that this method's name is a little misleading.